File: src/main/java/net/adventuregame/game/GameInit.java
// GameInit.java
package net.adventuregame.game;

import com.chidozie.core.renderEngine.WindowManager;
import net.adventuregame.items.Items;
import net.adventuregame.save.SaveManager;

/**
 * @version 1
 * @since 0.0.1
 * @author Chidozie Derek Chidozie-Uzowulu
 */
public class GameInit {
    private final GameState state;

    public GameInit(WindowManager window) {
        Items.registerAll();
        if (SaveManager.INSTANCE.getSaveFile().exists()) {
            this.state = SaveManager.INSTANCE.loadGame(window);
        } else {
            this.state = GameState.getInstance(window);
        }
    }

    /** Boot up every system in the correct order. */
    public void initializeAll() {
        state.initSystems();         // core: terrain, player, camera, renderer, picker
        state.loadFontsAndAudio();   // GUI text, audio, advancements
        state.initWorld();           // lights, trees, ferns, barrels, mobs, pickups
        state.initRenderers();       // GUI, water, particles, post-processing
    }

    public GameState getState() {
        return state;
    }
}

File: src/main/kotlin/net/adventuregame/game/GameInit.kt
// GameInit.java
package net.adventuregame.game

import com.adv.core.renderEngine.WindowManager
import net.adventuregame.game.GameState.Companion.getInstance
import net.adventuregame.items.Items
import net.adventuregame.save.SaveManager.loadGame
import net.adventuregame.save.SaveManager.saveFile

/**
 * @version 1
 * @since 0.0.1
 * @author Chidozie Derek Chidozie-Uzowulu
 */
class GameInit(window: WindowManager) {
    val state: GameState?

    init {
        Items.registerAll()
        if (saveFile.exists()) {
            this.state = loadGame(window)
        } else {
            this.state = getInstance(window)
        }
    }

    /** Boot up every system in the correct order.  */
    fun initializeGLStuff() {
        state!!.initSystems() // core: terrain, player, camera, renderer, picker
        state.loadFontsAndAudio() // GUI text, audio, advancements
    }
}
